interfaces/builtin/common_files.go

```diff
@@ -165,3 +166,65 @@ func (iface *commonFilesInterface) AppArmorConnectedPlug(spec *apparmor.Specific
 
 	return nil
 }
+
+// pathDependency calculates the path that must exist to allow creation of
+// the base element (directory or file) of the provided raw path. Supports
+// only paths that start with $HOME. In case of no valid dependency or
+// error an empty string is returned.
+func pathDependency(rawPath interface{}) (*mount.PathSpec, error) {
+	path, ok := rawPath.(string)
+	if !ok {
+		return nil, fmt.Errorf("%[1]v (%[1]T) is not a string", rawPath)
+	}
+
+	// The minimum viable path consists of three path elements, starting with
+	// directory $HOME and looks like $HOME/<dependant-dir-1>/<target-dir-or-file>
+	pathElements := strings.Split(path, string(filepath.Separator))
+	if len(pathElements) < 3 {
+		return nil, nil
+	}
+	if pathElements[0] != "$HOME" {
+		return nil, nil
+	}
+
+	pathSpec := &mount.PathSpec{
+		// Path that must exist and will be extended
+		// TODO: Should we expand $HOME?
```

Comment:
This looks like the place to do dir expansion:
https://github.com/snapcore/snapd/blob/master/cmd/snap-update-ns/user.go#L80-L82